libraries/core/src/main/java/com/paritytrading/philadelphia/FIXConnection.java

```diff
@@ -605,12 +605,16 @@ public void message(FIXMessage message) throws IOException {
         private void handleMsgSeqNum(FIXMessage message, FIXValue msgType, long msgSeqNum) throws IOException {
             if (msgSeqNum < rxMsgSeqNum)
                 handleTooLowMsgSeqNum(message, msgType, msgSeqNum);
-            else
+            else {
                 sendResendRequest(rxMsgSeqNum);
+                if (msgType.contentEquals(Logon)) {
+                    handleLogon(message);
+                }
+            }
         }
 
         private void handleTooLowMsgSeqNum(FIXMessage message, FIXValue msgType, long msgSeqNum) throws IOException {
-            if (msgType.contentEquals(Logout)) {
+            if (msgType.contentEquals(Logout) || msgType.contentEquals(Logon)) {
                 handleLogout(message);
```

Comment:
Should this perhaps look as follows?
```java
if (msgType.contentEquals(Logon)) {
    handleLogon(message);
} else if (msgType.contentEquals(Logout)) {
    handleLogout(message);
…
```

Reply:
Thank you for this suggestion @jvirtanen 

My idea is based on the FIX specs quoted here:  [Issuecomment](https://github.com/paritytrading/philadelphia/issues/315#issuecomment-1264348589)
But you are right, the Logon case should lead to a different call as the Logout case does, although both cases are Log**out** related.
How about this:

```java
if (msgType.contentEquals(Logon)) {
    sendLogout();
    statusListener.close(FIXConnection.this, "MsgSeqNum(34) too low");
} else if (msgType.contentEquals(Logout)) {
    handleLogout(message);
…

```

Reply:
Right, good point. I think the above would be a good approach; we already send Logout if the MsgSeqNum(34) field is missing. However, perhaps we could set the Text(58) field in the Logout(5) message also in this case:
```java
sendLogout("MsgSeqNum(34) too low");
```

Reply:
I am fine with the message in the Logout instead of the connection termination.
Can you confirm, we still want to close the connection after the Logout as stated in the FIX-docs? 

> followed by a termination of the transport layer connection

Reply:
Yes, let's have the `statusListener.close(...)` call there as you suggested above. 👍🏼